### PR TITLE
Add leading zeros support for SymbolsLongConverter.

### LINKED ISSUE #649: There's no way to encode a "zero" value in the SymbolsLongConverters
We take numeric zero to mean empty string which means strings that consist entirely of the "zero" character can't be round-tripped.

e.g. the following both fail
`assertEquals(Base85LongConverter.INSTANCE.asText(Base85LongConverter.INSTANCE.parse("0")), "0");`
`assertEquals(Base85LongConverter.INSTANCE.asText(Base85LongConverter.INSTANCE.parse("00000")), "00000");`

a similar problem exists for values with leading "zero" values e.g.
`assertEquals(Base85LongConverter.INSTANCE.asText(Base85LongConverter.INSTANCE.parse("00012")), "00012");`

Not sure if this is a bug, or just a limitation to be aware of when using them to encode strings. Perhaps it's something we need to be explicit about in our docs?

### PR DESCRIPTION
**Pull request by yevgenp**
Fixes #649

### CODE REVIEW

**Comment by yevgenp** [comment]
It's a possible implementation option which comes with its' own drawbacks (maxLength -1, long overflow).

**Comment by nicktindall** [comment]
Very clever. I like this approach better than https://github.com/OpenHFT/Chronicle-Wire/pull/822 because the latter still strips leading zeroes. If everyone's happy with the reduced max length, and we're certain that it's backwardly compatible then it's approved by me.

**Review by nicktindall — APPROVED** [review]

**Comment by nicktindall** [comment]
Be good to get @peter-k-lawrey's thoughts on this one

**Comment by JerryShea** [comment]
> Be good to get @peter-k-lawrey's thoughts on this one

What if he disagrees with @peter-lawrey ?

**Comment by nicktindall** [comment]
> > Be good to get @peter-k-lawrey's thoughts on this one
> 
> What if he disagrees with @peter-lawrey ?

Is there a hierarchy of Peters? surely they don't always agree

**Comment by JerryShea** [comment]
> > > Be good to get @peter-k-lawrey's thoughts on this one
> > 
> > 
> > What if he disagrees with @peter-lawrey ?
> 
> Is there a hierarchy of Peters? surely they don't always agree

Ask @peter-lawrey-admin - he surely is the boss

File: src/main/java/net/openhft/chronicle/wire/converter/SymbolsLongConverter.java [hunk 1]
```diff
@@ -149,6 +154,9 @@ public void append(Bytes<?> text, long value) {
             text.append(decode[v]);
         }
 
+        if (!text.isEmpty())
+            text.unwrite(text.writePosition() - 1, 1);
+
```
Comment (nicktindall) [inline src/main/java/net/openhft/chronicle/wire/converter/SymbolsLongConverter.java:159]: Actually this isn't backward or forward compatible. I really like it though, perhaps we make it a separate implementation and deprecate this one....

e.g. when encoded by the new impl and read by the old impl you get
`5HELLO`

and when encoded by the old impl and read by the new impl you get:
`ELLO`

**Review by nicktindall — CHANGES_REQUESTED** [review]
We should make this a separate implementation and deprecate the existing implementation, or even keep them as alternatives. It's not forward or backward coimpatible which is an issue if you have persisted data, or if you upgrade services that communicate independently.

**Comment by yevgenp** [comment]
Already implemented in #822

**Pull request closed without merging**